File: python_leaoai/python_leaoai/services/ApplicationServices/Pages/HeadContentService.py
from typing import TypeVar, Optional

T = TypeVar('T')

# --- Model equivalente ---
class Head:
    def __init__(self, page_title: Optional[str] = None,
                 meta_description: Optional[str] = None,
                 image_open_graph: Optional[str] = None):
        self.page_title = page_title
        self.meta_description = meta_description
        self.image_open_graph = image_open_graph

class HeadContentService:
    def build(self, obj: T) -> Head:
        """
        Constrói um Head a partir de qualquer objeto que tenha as propriedades 'page_title', 'meta_description' e 'meta_image'.
        """
        page_title = self.get_property_value(obj, "page_title")
        meta_description = self.get_property_value(obj, "meta_description")
        meta_image = self.get_property_value(obj, "meta_image")
        image_open_graph = meta_image  # Mantendo lógica do C#

        return Head(
            page_title=page_title,
            meta_description=meta_description,
            image_open_graph=image_open_graph
        )

    def get_property_value(self, obj: T, property_name: str):
        """
        Retorna o valor de uma propriedade do objeto ou None se não existir.
        """
        return getattr(obj, property_name, None)

File: Framework.Services/ApplicationServices/Pages/HeadContentService.cs
using Framework.Domain.Site.Core;
using System.Reflection;

namespace Framework.Services.ApplicationServices.Pages
{
    public class HeadContentService
    {
        public Head Build<T>(T obj)
        {
            // Usando reflexão para obter os valores das propriedades
            var pageTitle = GetPropertyValue<string>(obj, "PageTitle");
            var metaDescription = GetPropertyValue<string>(obj, "MetaDescription");
            var metaImage = GetPropertyValue<string>(obj, "MetaImage");
            var imageOpenGraph = GetPropertyValue<string>(obj, "MetaImage");

            return new Head()
            {
                PageTitle = pageTitle,
                ImageOpenGraph = imageOpenGraph,
                MetaDescription = metaDescription
            };
        }

        private TValue GetPropertyValue<TValue>(object obj, string propertyName)
        {
            // Obtém o tipo do objeto
            var type = obj.GetType();

            // Obtém a propriedade pelo nome
            var property = type.GetProperty(propertyName, BindingFlags.Public | BindingFlags.Instance);

            if (property == null || !typeof(TValue).IsAssignableFrom(property.PropertyType))
            {
                return default;
            }

            // Retorna o valor da propriedade
            return (TValue)property.GetValue(obj);
        }
    }
}
